src/plugins/action/action_impl.cpp

```diff
@@ -389,19 +389,40 @@ void ActionImpl::goto_location_async(
     const float yaw_deg,
     const Action::ResultCallback& callback)
 {
-    MavlinkCommandSender::CommandInt command{};
-
-    command.command = MAV_CMD_DO_REPOSITION;
-    command.target_component_id = _parent->get_autopilot_id();
-    command.params.param4 = to_rad_from_deg(yaw_deg);
-    command.params.x = int32_t(std::round(latitude_deg * 1e7));
-    command.params.y = int32_t(std::round(longitude_deg * 1e7));
-    command.params.z = altitude_amsl_m;
+    auto send_do_reposition =
+        [this, callback, yaw_deg, latitude_deg, longitude_deg, altitude_amsl_m]() {
+            MavlinkCommandSender::CommandInt command{};
+
+            command.command = MAV_CMD_DO_REPOSITION;
+            command.target_component_id = _parent->get_autopilot_id();
+            command.params.param4 = to_rad_from_deg(yaw_deg);
+            command.params.x = int32_t(std::round(latitude_deg * 1e7));
+            command.params.y = int32_t(std::round(longitude_deg * 1e7));
+            command.params.z = altitude_amsl_m;
+
+            _parent->send_command_async(
+                command, [this, callback](MavlinkCommandSender::Result result, float) {
+                    command_result_callback(result, callback);
+                });
+        };
+
+    // Change to Hold mode first
+    if (_parent->get_flight_mode() != SystemImpl::FlightMode::Hold) {
+        _parent->set_flight_mode_async(
+            SystemImpl::FlightMode::Hold,
+            [callback, send_do_reposition](MavlinkCommandSender::Result result, float) {
+                Action::Result action_result = action_result_from_command_result(result);
+                if (action_result != Action::Result::Success) {
+                    if (callback) {
+                        callback(action_result);
+                    }
```

Comment:
Actually, I think it can be done just like in the other places in action_impl.cpp:

```suggestion
                    command_result_callback(callback, action_result);
```

Reply:
So why is it different for the `arm` command? Is it because the API was just not updated?

Reply:
I would say so, but let's ask @julianoes to have a second opinion :)

Reply:
I think this is also wrong for the `arm` command. The point is that `call_user_callback` needs to be used which happens in `command_result_callback`.

Reply:
Done